hw/top_earlgrey/syn/chip_earlgrey_asic.ot.sdc

```diff
@@ -0,0 +1,1570 @@
+    -to [get_clocks SPI_DEV_FAST_PASS_CLK] -through [get_ports ${SPI_DEV_DATA_PORTS}]
+
+# bidir ports facing storage device
+if {$spec_constr} {
+set_input_delay -min ${SPI_HOST_IN_DEL_MIN} ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_input_delay -max ${SPI_HOST_IN_DEL_MAX} ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_output_delay -min ${SPI_HOST_OUT_DEL_MIN} \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_output_delay -max ${SPI_HOST_OUT_DEL_MAX} \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_FAST_PASS_CLK -add_delay
+} else {
+set_input_delay -min $spi_fast_pass_host_inp_min ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_input_delay -max $spi_fast_pass_host_inp_max ${SPI_HOST_DATA_PORTS} \
+    -clock_fall -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_output_delay -min $spi_fast_pass_flsh_out_min \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_FAST_PASS_CLK -add_delay
+set_output_delay -max $spi_fast_pass_flsh_out_max \
+    [get_ports "SPI_HOST_CS_L ${SPI_HOST_DATA_PORTS}"] \
+    -clock SPI_HOST_FAST_PASS_CLK -add_delay
+}
+# Fast commands must use the fast read pipeline. Disable timing for the
+# combinatorial passthrough path.
+set_false_path -from [get_clocks SPI_HOST_FAST_PASS_CLK] \
+    -through [get_ports ${SPI_DEV_DATA_PORTS}] \
+    -to [get_clocks SPI_DEV_FAST_PASS_CLK]
+
+#leonids 03/06/2024 Update from Alex
+# full-cycle sampling flops
+set_false_path -from [get_clocks SPI_HOST_FAST_PASS_CLK] \
+    -to [get_clocks SPI_DEV_FAST_PASS_IN_CLK]
+
+# CSB clock from top-level port (added to SPI_DEV_CSB_CLK)
+create_clock -name SPI_DEV_FAST_PASS_CSB_CLK -add \
+    -period [expr 2 * ${SPI_DEV_FAST_PASS_TCK}] \
+    -waveform "${SPI_DEV_FAST_PASS_TCK_HALF} [expr ${SPI_DEV_FAST_PASS_TCK_HALF} + ${SPI_DEV_FAST_PASS_TCK}]" \
+    [get_ports SPI_DEV_CS_L]
+if {$spec_constr} {
+set_clock_latency -source -min ${SPI_DEV_IN_DEL_MIN} \
+    -clock SPI_DEV_FAST_PASS_CSB_CLK [get_ports SPI_DEV_CS_L]
+set_clock_latency -source -max ${SPI_DEV_IN_DEL_MAX} \
+    -clock SPI_DEV_FAST_PASS_CSB_CLK [get_ports SPI_DEV_CS_L]
+} else {
+set_clock_latency -source -min $spi_dev_inp_csb_min \
+    -clock SPI_DEV_FAST_PASS_CSB_CLK [get_ports SPI_DEV_CS_L]
+set_clock_latency -source -max $spi_dev_inp_csb_max \
+    -clock SPI_DEV_FAST_PASS_CSB_CLK [get_ports SPI_DEV_CS_L]
+}
+set_propagated_clock [get_clock SPI_DEV_FAST_PASS_CSB_CLK]
+
+# CSB-clocked status bits to various negedge-triggered flops, especially in the
+# serializer.
+# Advance the hold edge by one cycle, since CSB changes nominally on the same
+# edge as SPI_DEV_FAST_PASS_OUT_CLK, but SPI_DEV_FAST_PASS_OUT_CLK isn't
+# actually toggling.
+#set_ideal_network [get_pins top_earlgrey/u_spi_device/u_csb_rst_scan_mux/clk_o]
```

Comment:
This constraint is commented out, should it be removed instead?

Reply:
This commented out constraint is still part of the last revision. It's fine to leave them in to not further increase the size of the diff (these comments were already part of the SDC used for ES).